packages/synapse-react-client/src/components/EntityFinder/tree/VirtualizedTree.tsx

```diff
@@ -523,27 +538,44 @@ export const VirtualizedTree = (props: VirtualizedTreeProps) => {
     // Because we update the root node with a copy at the end of this function, we can write to the node under update.
     async (node: Writable<EntityHeaderNode>) => {
       // Fetch the children of the node
-      const children = await SynapseClient.getEntityChildren(
-        {
-          parentId: node.id,
-          nextPageToken: node.childrenNextPageToken,
-          includeTypes: visibleTypes,
-        },
-        accessToken,
-      )
-
+      const entityChildrenRequest: EntityChildrenRequest = {
+        parentId: node.id,
+        nextPageToken: node.childrenNextPageToken,
+        includeTypes: visibleTypes,
+      }
+      const children = await queryClient.fetchQuery({
+        queryKey: keyFactory.getEntityChildrenQueryKey(
+          entityChildrenRequest,
+          false,
+        ),
+        queryFn: () => getEntityChildren(entityChildrenRequest, accessToken),
+      })
       // Update the node data -- add the children and store the nextPageToken
       if (node.children) {
-        node.children.push(...children.page)
+        for (const newChild of children.page) {
+          // Prevent adding duplicate entries
+          if (!node.children.find(child => child.id == newChild.id)) {
+            node.children.push(newChild)
+          }
+        }
       } else {
         node.children = children.page
       }
       node.childrenNextPageToken = children.nextPageToken
 
+      // Ensure the node for which we fetched children is expanded/open
+      if (treeInstance.current) {
+        await treeInstance.current.recomputeTree({
+          [node.id]: {
+            open: true,
+          },
+        })
+      }
```

Comment:
I think there's another issue where sometimes the tree doesn't expand all the way after data is fetched. This should make that more reliable